README.md

```diff
@@ -10,218 +10,326 @@
+
+<br>
+
+## Build Options
+### Considerations
+- The **`kvssink`** GStreamer plugin and samples, and the **JNI** are _not_ built by default. To build them, include their corresponding cmake command arguments: `cmake .. -DBUILD_GSTREAMER_PLUGIN=ON -DBUILD_JNI=TRUE`
+- By default, the **dependency libraries** are installed from GitHub and built locally. To instead link to pre-installed libraries on the device, include the following cmake command argument: `cmake .. -DBUILD_DEPENDENCIES=OFF`
+  - The dependency libraries are Curl, OpenSSL, and Log4Cplus.
+ 
+### CMake Arguments
+You can pass the following additional CMake options:
+
+| Option	                     | Default       | Description	|
+|:-----------------------------|:-------------:|:-------------|
+| BUILD_GSTREAMER_PLUGIN       | OFF           | Build the `kvssink` GStreamer plugin
+| BUILD_JNI                    | OFF           | Build C++ wrapper for JNI to expose the functionality to Java/Android
+| BUILD_DEPENDENCIES           | ON            | Build depending libraries from source
+| BUILD_TEST                   | OFF           | Build unit/integration tests, may be useful to confirm support for your device, to run tests:       `./tst/producerTest`
+| CODE_COVERAGE                | OFF           | Enable coverage reporting
+| COMPILER_WARNINGS            | OFF           | Enable all compiler warnings
+| ADDRESS_SANITIZER            | OFF           | Build with AddressSanitizer
+| MEMORY_SANITIZER             | OFF           | Build with MemorySanitizer
+| THREAD_SANITIZER             | OFF           | Build with ThreadSanitizer
+| UNDEFINED_BEHAVIOR_SANITIZER | OFF           | Build with UndefinedBehaviorSanitizer
+| ALIGNED_MEMORY_MODEL         | OFF           | Build for aligned memory model only devices
+| BUILD_LOG4CPLUS_HOST         | OFF           | Specify host-name for log4cplus for cross-compilation
+| CONSTRAINED_DEVICE           | OFF           | Set the thread stack size to 0.5MB, needed for Alpine builds
+
+These options can be set as arguments to the `cmake` command, for example:
+```
+cmake .. -DBUILD_GSTREAMER_PLUGIN=ON -DBUILD_DEPENDENCIES=OFF
 ```
 
-If the build failed, or `GST_PLUGIN_PATH` is not properly set you will get output like
+### Setting the Log Level
+To set a log level, update the log level value [here](https://github.com/awslabs/amazon-kinesis-video-streams-producer-sdk-cpp/blob/master/kvs_log_configuration#L1).
 
-```text
-No such element or plugin 'kvssink'
-```
+The log levels currently available with `log4cplus` are:
+1. `TRACE`
+2. `DEBUG` 
+3. `INFO`   
+4. `WARN`   
+5. `ERROR`   
+6. `FATAL`   
```

Comment:
Shouldn't there also be SILENT level?
https://github.com/awslabs/amazon-kinesis-video-streams-producer-sdk-cpp/issues/699

Reply:
The Log4Cplus logger we use for this SDK does not have SILENT level.